backend/src/nodes/properties/outputs/numpy_outputs.py

```diff
@@ -96,10 +96,18 @@ def get_broadcast_data(self, value: np.ndarray):
         img = value
         h, w, c = get_h_w_c(img)
 
-        base64_img = preview_encode(img, 512)
+        # Encode for multiple scales. Use the preceding scale to save time encoding the smaller sizes.
+        base64_2048, np_2048 = preview_encode(img, 2048)
```

Comment:
I don't like that we PNG encode small images multiple times. Basically, if the image is smaller than the next smaller preview size, then we don't need that preview (e.g. if the image is 500px, then we don't need >=1024px).

Of course, variably many previews doesn't work with the hard-coded preview sizes, so we should probably replace them with a `preview: { size: number, url: string }[]` prop.

Reply:
Also, since `preview_encode` treats `target_size` as more of a suggestion, `size: number` should be the actual size of the (returned) preview image.

Reply:
How will the frontend decide which to show then? Index?

Reply:
It knows the size of each preview. That's enough information to pick one.